python/demo/demo_mixed-poisson-bddc.py

```diff
@@ -0,0 +1,166 @@
+# types
+dtype = dolfinx.default_scalar_type
+xdtype = dolfinx.default_real_type
+# -
+
+# Create a two-dimensional mesh. The iterative solver supports three-
+# dimensional meshes too. BDDC methods needs the linear operator
+# to be stored in unassembled format, i.e. assembled on each MPI process
+# associated with the subdomain part of the mesh, but not across processes
+# To this end, we require the mesh to not be distributed with overlap
+
+# +
+msh = create_unit_square(MPI.COMM_WORLD, 96, 96, CellType.triangle, ghost_mode=GhostMode.none, dtype=xdtype)
+# -
+
+# From now on, the script is the same as demo_mixed-poisson.py.
+
+# +
+k = 1
+V = fem.functionspace(msh, element("RT", msh.basix_cell(), k, dtype=xdtype))
+W = fem.functionspace(msh, element("Discontinuous Lagrange", msh.basix_cell(), k - 1, dtype=xdtype))
+Q = ufl.MixedFunctionSpace(V, W)
+
+sigma, u = ufl.TrialFunctions(Q)
+tau, v = ufl.TestFunctions(Q)
+
+x = ufl.SpatialCoordinate(msh)
+f = 10 * ufl.exp(-((x[0] - 0.5) * (x[0] - 0.5) + (x[1] - 0.5) * (x[1] - 0.5)) / 0.02)
+
+dx = ufl.Measure("dx", msh)
+
+a = ufl.extract_blocks(
+    ufl.inner(sigma, tau) * dx + ufl.inner(u, ufl.div(tau)) * dx + ufl.inner(ufl.div(sigma), v) * dx
+)
+L = [ufl.ZeroBaseForm((tau,)), -ufl.inner(f, v) * dx]
+
+fdim = msh.topology.dim - 1
+facets_top = mesh.locate_entities_boundary(msh, fdim, lambda x: np.isclose(x[1], 1.0))
+facets_bottom = mesh.locate_entities_boundary(msh, fdim, lambda x: np.isclose(x[1], 0.0))
+dofs_top = fem.locate_dofs_topological(V, fdim, facets_top)
+dofs_bottom = fem.locate_dofs_topological(V, fdim, facets_bottom)
+
+cells_top_ = mesh.compute_incident_entities(msh.topology, facets_top, fdim, fdim + 1)
+cells_bottom = mesh.compute_incident_entities(msh.topology, facets_bottom, fdim, fdim + 1)
+g = fem.Function(V, dtype=dtype)
+g.interpolate(lambda x: np.vstack((np.zeros_like(x[0]), np.sin(5 * x[0]))), cells0=cells_top_)
+g.interpolate(lambda x: np.vstack((np.zeros_like(x[0]), -np.sin(5 * x[0]))), cells0=cells_bottom)
+bcs = [fem.dirichletbc(g, dofs_top), fem.dirichletbc(g, dofs_bottom)]
+
+sigma, u = fem.Function(V, name="sigma", dtype=dtype), fem.Function(W, name="u", dtype=dtype)
+# -
+
+# We now create a linear problem solver for the mixed problem.
+# As we will use BDDC, we specify the matrix kind to be "is".
+# We also show a minimal customization for the method to be
+# applied monolithically on the symmetric indefinite linear
+# system arising from the mixed problem.
+# In particular, we need solvers for the subproblems that can
+# support the saddle point formulation.
+
+# +
+if PETSc.Sys().hasExternalPackage("umfpack"):
+    solver = "lu"
+    solver_type = "umfpack"
+elif PETSc.Sys().hasExternalPackage("mumps"):
+    solver = "cholesky"
+    solver_type = "mumps"
+elif PETSc.Sys().hasExternalPackage("superlu"):
+    solver = "lu"
+    solver_type = "superlu"
+elif PETSc.Sys().hasExternalPackage("umfpack"):
```

Comment:
Duplicate UMFPACK check? Also, why would you not default to MUMPS/Cholesky?

Reply:
will fix, thanks